### PR TITLE
fix: several bug fixes

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
https://mask.atlassian.net/browse/MF-1819
https://mask.atlassian.net/browse/MF-1783
https://mask.atlassian.net/browse/MF-1805
https://mask.atlassian.net/browse/MF-969

image

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Review by UncleBill — DISMISSED** [review]

File: packages/mask/src/social-network-adaptor/facebook.com/injection/Toolbar.tsx [hunk 1]
```diff
@@ -1,13 +1,21 @@
 import { MutationObserverWatcher } from '@dimensiondev/holoflows-kit'
 import { createReactRootShadowed } from '../../../utils/shadow-root/renderInShadowRoot'
 import { startWatch } from '../../../utils/watcher'
-import { toolBoxInSideBarSelector } from '../utils/selector'
+import { toolBoxInSideBarSelector, toolBoxInSideBarSelectorWithNoLeftRailStart } from '../utils/selector'
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/facebook.com/injection/Toolbar.tsx]:
```suggestion
import { toolboxInSidebarSelector, toolboxInSidebarSelectorWithNoLeftRailStart } from '../utils/selector'
```

File: packages/web3-shared/base/src/specs/index.ts [hunk 2]
```diff
@@ -522,6 +522,10 @@ export interface TransactionDescriptor<ChainId, Transaction> {
     successfulDescription?: string
     /** a human-readable description for failed transaction. */
     failedDescription?: string
+    /** The address of the token leveraged to swap other tokens */
+    tokenInAddress?: string
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/specs/index.ts:526]: From the comments, I found we probably violet some design principles here.

> **Don't hoist subclass fields to super-class. It breaks the hierarchy between classes.**

What about using a more abstract concept? We need it must be reasonable for all transactions.

File: packages/mask/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx [hunk 3]
```diff
@@ -152,11 +152,17 @@ const ContractInteraction = memo(() => {
...
+    const { Others, TransactionFormatter } = useWeb3State(NetworkPluginID.PLUGIN_EVM)
     const chainId = useChainId(NetworkPluginID.PLUGIN_EVM)
     const networkType = useNetworkType(NetworkPluginID.PLUGIN_EVM)
     const [transferError, setTransferError] = useState(false)
     const { value: request, loading: requestLoading } = useUnconfirmedRequest()
+    const { value: transactionDescription } = useAsync(async () => {
+        return TransactionFormatter?.formatTransaction?.(request?.transactionContext?.chainId!, {
+            ...request?.transactionContext!,
+            data: request?.computedPayload?.data,
+        })
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx]: In a safer way and no fighting with typings.

```suggestion
        if (!request?.transactionContext?.chainId) return
        return TransactionFormatter?.formatTransaction?.(request.transactionContext.chainId, {
            ...request.transactionContext,
            data: request?.computedPayload?.data,
        })
```

File: packages/mask/src/social-network-adaptor/facebook.com/injection/Toolbar.tsx [hunk 4]
```diff
@@ -1,13 +1,21 @@
 import { MutationObserverWatcher } from '@dimensiondev/holoflows-kit'
 import { createReactRootShadowed } from '../../../utils/shadow-root/renderInShadowRoot'
 import { startWatch } from '../../../utils/watcher'
-import { toolBoxInSideBarSelector } from '../utils/selector'
+import { toolboxInSideBarSelector, toolboxInSideBarSelectorWithNoLeftRailStart } from '../utils/selector'
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/facebook.com/injection/Toolbar.tsx]:
```suggestion
import { toolboxInSidebarSelector, toolboxInSidebarSelectorWithNoLeftRailStart } from '../utils/selector'
```

File: packages/mask/src/social-network-adaptor/facebook.com/utils/selector.ts [hunk 5]
```diff
@@ -98,14 +98,18 @@ export const searchFacebookConfirmAvatarImageSelector = () =>
...
-export const toolBoxInSideBarSelector: () => LiveSelector<E, true> = () =>
+export const toolboxInSideBarSelector: () => LiveSelector<E, true> = () =>
     /* cspell:disable-next-line */
     querySelector<E>('#ssrb_left_rail_start')
         .closest(1)
         .querySelector('h2')
         .closest(1)
         .querySelector('div > div > div > :nth-child(2) > ul > li:nth-child(2)')
 
+export const toolboxInSideBarSelectorWithNoLeftRailStart: () => LiveSelector<E, true> = () =>
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/facebook.com/utils/selector.ts]:
```suggestion
export const toolboxInSidebarSelectorWithNoLeftRailStart: () => LiveSelector<E, true> = () =>
```

File: packages/mask/src/social-network-adaptor/facebook.com/utils/selector.ts [hunk 6]
```diff
@@ -98,14 +98,18 @@ export const searchFacebookConfirmAvatarImageSelector = () =>
 export const inpageAvatarSelector = () => querySelectorAll('[type="nested/pressable"]').closest<HTMLAnchorElement>(2)
 // #region
 
-export const toolBoxInSideBarSelector: () => LiveSelector<E, true> = () =>
+export const toolboxInSideBarSelector: () => LiveSelector<E, true> = () =>
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/facebook.com/utils/selector.ts]:
```suggestion
export const toolboxInSidebarSelector: () => LiveSelector<E, true> = () =>
```

File: packages/plugins/EVM/src/index.ts [hunk 7]
```diff
@@ -52,3 +52,5 @@ registerPlugin<
             import.meta.webpackHot && import.meta.webpackHot.accept('./Worker', () => hot(import('./Worker'))),
     },
 })
+
+export { TransactionFormatter } from './state'
```
Comment (guanbinrui) [inline packages/plugins/EVM/src/index.ts]: Strange 🤔

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**